scripts/tpg_find_overactive_channels.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+"""
+For a given TPStream, plot the channel histogram and print
+the channels that are above a given limit and Tuckey's fence.
+"""
+
+
+from trgtools import TPReader
+
+import click
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+import os
+
+
+# As of 05-13-2024.
+DETECTOR_CHANNELMAPS = {
+            "HD": os.path.join(os.environ["DETCHANNELMAPS_SHARE"], "config/pd2hd/PD2HDChannelMap_v6.txt"),
+            "VD": os.path.join(os.environ["DETCHANNELMAPS_SHARE"], "config/pd2vd/PD2VDBottomTPCChannelMap_v1.txt"),
+            "VDCB": os.path.join(os.environ["DETCHANNELMAPS_SHARE"], "config/vdcoldbox/vdcbce_chanmap_v4.txt")
+        }
+
+# Colors.
+BLUE = "#63ACBE"
+RED  = "#EE442F"
+
+
+def get_detector_element_channel_hist_map(map_name: str) -> dict[str, int]:
+    """
+    Get a map with the detector elements as the key and the channels (as histogram edges)
+    as the value.
+
+    Parameters:
+        map_name (str): Channel map name to load from.
+
+    Returns a dict with the detector element name and channel histogram edges.
+    """
+    # Only care about the offline channels (0) and detector element name (2) columns.
+    df = pd.read_table(DETECTOR_CHANNELMAPS[map_name], sep="\s+", usecols=[0, 2], header=None)
+    named_series = pd.Series(df[0].values, index=df[2])  # Many channels will belong to the same det_elem.
+    unique_det_elems = named_series.keys().unique()      # So the det_elem will repeat many times.
+
+    det_elem_edges = {}
+    for det_elem in unique_det_elems:
+        min_channel = named_series[det_elem].values.min()
+        max_channel = named_series[det_elem].values.max()
+        det_elem_edges[det_elem] = np.arange(min_channel-0.5, max_channel+1)  # Need padding for hist edges.
+
+    return det_elem_edges
+
+
+def read_requested_fragments(reader: TPReader, fragment_index: int, num_fragments: int) -> None:
+    """
+    Read the requested fragments and make changes if necessary.
+
+    Parameters:
+        reader          (TPReader): TPReader object to read fragments from.
+        fragment_index  (int)     : First fragment path to read from.
+        num_fragments   (int)     : Number of fragments to read.
+
+    Returns nothing. Mutates :reader:.
+    """
+    len_paths = len(reader.get_fragment_paths())
+    if fragment_index > len_paths:
+        raise IndexError(f"There are only {len_paths} fragment paths. Choose a smaller index.")
+    if fragment_index + num_fragments > len_paths:
+        num_fragments = len_paths - fragment_index
+        print( "WARNING: fragment_index + num_fragments > the number of fragments in the file.")
+        print(f"         Using num_fragments = {num_fragments} instead.")
```

Comment:
I was going to ask for `num_fragments = -1` to correspond to all the fragments in the file, but this line resolves it!
I imagine if we want to process all the fragments, then we can simply supply `fragment_index = 0` and set `num_fragments` to something obscene like 1e9?

Reply:
Definitely, that would work, but it would probably be better to have a `-a|--all` option. I can add this in the future. It may be slow to use if the TriggerRecords or TimeSlices are large though.